spyder_unittest/backend/testrunner.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Developers
+#
+# Licensed under the terms of the MIT License
+# (see LICENSE.txt for details)
+# -----------------------------------------------------------------------------
+"""Classes for running tests within various frameworks."""
+
+# Standard library imports
+from collections import namedtuple
+import os
+import tempfile
+
+# Third party imports
+from lxml import etree
+from qtpy.QtCore import (QObject, QProcess, QProcessEnvironment, QTextCodec,
+                         Signal)
+from spyder.py3compat import to_text_string
+from spyder.utils.misc import add_pathlist_to_PYTHONPATH
+
+# Class representing test results
+TestResult = namedtuple('TestResult', [
+    'category', 'status', 'name', 'message', 'time', 'extra_text'
+])
+
+
+class Category:
+    """Enum type representing category of test result."""
+
+    OK = 1
+    FAIL = 2
+    SKIP = 3
```

Comment:
Just a thought not a review comment: If we plan to do more fancy stuff with this like iterating, we might considering using enum from the standard library (py>34) or https://pypi.python.org/pypi/enum34 (py<34).